spring-cloud-gcp-data-firestore/src/main/java/com/google/cloud/spring/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -216,9 +220,19 @@ public Object execute(Object[] parameters) {
 							"Too few parameters are provided for query method: " + getQueryMethod().getName());
 				}
 				Object value = it.next();
-				filter.getFieldFilterBuilder().setField(fieldReference)
-						.setOp(getOperator(part, value))
-						.setValue(this.classMapper.toFirestoreValue(value));
+				if (fieldReference.getFieldPath().equals(FieldPath.documentId().toString())) {
+					FirestoreTemplate firestoreTemplate = (FirestoreTemplate) this.reactiveOperations;
+					DocumentReference documentReference = firestoreTemplate.getDocumentReference(persistentEntity.collectionName(), (String) value);
+					filter.getFieldFilterBuilder().setField(fieldReference)
+							.setOp(getOperator(part, documentReference))
+							.setValue(Value.newBuilder().setReferenceValue(
+									firestoreTemplate.buildResourceName(persistentEntity, documentReference.getId())));
```

Comment:
Here again, we know what the *value* of the `@DocumentId` field is without having to look it up from document reference. It's the `value` local variable!
Now, the local variable is an `Object`, while you need a `String`. But per our [documentation](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/docs/src/main/asciidoc/firestore.adoc#object-mapping), `@DocumentId` can only be applied to a `String` field since that's what Firestore uses for document IDs.

Reply:
That is a great point! I cleaned this up a bit to use `value` directly.